rllib/evaluation/rollout_worker.py

```diff
@@ -1182,16 +1200,31 @@ def set_policy_mapping_fn(
                 raise ValueError("`policy_mapping_fn` must be a callable!")
 
     @DeveloperAPI
-    def set_policies_to_train(
-            self, policies_to_train: Optional[List[PolicyID]] = None) -> None:
-        """Sets `self.policies_to_train` to a new list of PolicyIDs.
+    def set_is_policy_to_train(
+            self,
+            is_policy_to_train: Optional[Union[Container[PolicyID], Callable[
+                [PolicyID, SampleBatchType], bool]]] = None) -> None:
+        """Sets `self.policies_to_train` to a new set or callable.
 
         Args:
-            policies_to_train: The new list of policy IDs to train with.
-                If None, will keep the existing list in place.
+            is_policy_to_train: An optional container of policy IDs to be
+                trained or a callable taking PolicyID and - optionally -
+                SampleBatchType and returning a bool (trainable or not?).
+                If None, will keep the existing setup in place.
+                Policies, whose IDs are not in the list (or for which the
+                callable returns False) will not be updated.
         """
-        if policies_to_train is not None:
-            self.policies_to_train = policies_to_train
+        if is_policy_to_train is not None:
```

Comment:
should None reset self.is_policy_to_train to empty list?
I also feel like we shouldn't allow is_policy_to_train to have a default value, which does nothing. e.g., why would someone do
```
self.set_is_policy_to_train()
```
by itself ... ?

Reply:
Hmm, actually, you are right. Not sure why this should be supported. I'll fix.